be/src/block_cache/block_cache.h

```diff
@@ -0,0 +1,47 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include "block_cache/kv_cache.h"
+#include "common/status.h"
+
+namespace starrocks {
+
+class BlockCache {
+public:
+    typedef std::string CacheKey;
+
+    ~BlockCache();
+
+    // Return a singleton block cache instance
+    static BlockCache* instance();
+
+    // Init the block cache instance
+    Status init(const CacheOptions& options);
+
+    // Set block size as the cache unit
+    Status set_block_size(size_t block_size);
+
+    // Write data to cache, the offset must be aligned by block size
+    Status write_cache(const CacheKey& cache_key, off_t offset, size_t size, const char* buffer,
+                       size_t ttl_seconds = 0);
+
+    // Read data from cache, it returns the data size if successful; otherwise the error status
+    // will be returned. The offset and size must be aligned by block size.
+    StatusOr<size_t> read_cache(const CacheKey& cache_key, off_t offset, size_t size, char* buffer);
+
+    // NOTICE: This function is not safe now, as the returned buffer may be evicted before visited
+    // by users. We need to implement it more safe by cachelib item handle.
+    Status read_cache_zero_copy(const CacheKey& cache_key, off_t offset, size_t size, const char** buf);
```

Comment:
consider return a wrap class instead of vanilla buf. such that you could control the life time of the cache object

Reply:
> 

Yes，a wrap class like baidu `IOBuf` can keep the buffer life time by reference count and avoid copying raw content. Another benefit is that it can represent a data target composed of some discontinuous buffer, when reading a large data, it can avoid copying different cache block content to a single buffer. We will consider to update it in the next version.